yb-voyager/cmd/analyzeSchema.go

```diff
@@ -686,9 +688,97 @@ var UnsupportedIndexDatatypes = []string{
 	"path",
 	"polygon",
 	"txid_snapshot",
+	"daterange",
+	"tsrange",
+	"tstzrange",
+	"numrange",
+	"int4range",
+	"int8range",
+	"interval", // same for INTERVAL YEAR TO MONTH and INTERVAL DAY TO SECOND
 	// array as well but no need to add it in the list as fetching this type is a different way TODO: handle better with specific types
 }
 
+func reportUnsupportedConstraintsOnComplexDatatypesInAlter(alterTableNode *pg_query.Node_AlterTableStmt, sqlStmtInfo sqlInfo, fpath string) {
+	schemaName := alterTableNode.AlterTableStmt.Relation.Schemaname
+	tableName := alterTableNode.AlterTableStmt.Relation.Relname
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	alterCmd := alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd()
+	unsupportedColumnsForTable, ok := columnsWithUnsupportedIndexDatatypes[fullyQualifiedName]
+	if !ok {
+		return
+	}
+	if alterCmd.GetSubtype() != pg_query.AlterTableType_AT_AddConstraint {
+		return
+	}
+	if !slices.Contains([]pg_query.ConstrType{pg_query.ConstrType_CONSTR_PRIMARY, pg_query.ConstrType_CONSTR_UNIQUE},
+		alterCmd.GetDef().GetConstraint().GetContype()) {
+		return
+	}
+	columns := alterCmd.GetDef().GetConstraint().GetKeys()
+	for _, col := range columns {
+		colName := col.GetString_().Sval
+		typeName, ok := unsupportedColumnsForTable[colName]
+		if ok {
+			displayName := fmt.Sprintf("%s, constraint: %s", fullyQualifiedName, alterCmd.GetDef().GetConstraint().GetConname())
+			reportCase(fpath, fmt.Sprintf(ISSUE_CONSTRAINT_WITH_COMPLEX_DATATYPES, typeName), "https://github.com/yugabyte/yugabyte-db/issues/9698",
```

Comment:
the link only points to citext not being supported? Should we be creating a separate issue for this?

Reply:
yeah created a new generic ticket and added that in all cases